plugins/utils/hardtime.nix

```diff
@@ -1,244 +1,204 @@
-        "2" = [
-          "n"
-          "x"
-        ];
-        "3" = [
-          "n"
-          "x"
-        ];
-        "4" = [
-          "n"
-          "x"
-        ];
-        "5" = [
-          "n"
-          "x"
-        ];
-        "6" = [
-          "n"
-          "x"
-        ];
-        "7" = [
-          "n"
-          "x"
-        ];
-        "8" = [
-          "n"
-          "x"
-        ];
-        "9" = [
-          "n"
-          "x"
-        ];
-        "c" = [ "n" ];
-        "C" = [ "n" ];
-        "d" = [ "n" ];
-        "x" = [ "n" ];
-        "X" = [ "n" ];
-        "y" = [ "n" ];
-        "Y" = [ "n" ];
-        "p" = [ "n" ];
-        "P" = [ "n" ];
-      } "Keys in what modes that reset the count.";
-
-      restrictedKeys = helpers.defaultNullOpts.mkAttrsOf (with types; listOf str) {
-        "h" = [
-          "n"
-          "x"
-        ];
-        "j" = [
-          "n"
-          "x"
-        ];
-        "k" = [
-          "n"
-          "x"
-        ];
-        "l" = [
-          "n"
-          "x"
-        ];
-        "-" = [
-          "n"
-          "x"
-        ];
-        "+" = [
-          "n"
-          "x"
-        ];
-        "gj" = [
-          "n"
-          "x"
-        ];
-        "gk" = [
-          "n"
-          "x"
-        ];
-        "<CR>" = [
-          "n"
-          "x"
-        ];
-        "<C-M>" = [
-          "n"
-          "x"
-        ];
-        "<C-N>" = [
-          "n"
-          "x"
-        ];
-        "<C-P>" = [
-          "n"
-          "x"
-        ];
-      } "Keys in what modes triggering the count mechanism.";
-
-      restrictionMode =
-        helpers.defaultNullOpts.mkEnumFirstDefault
-          [
-            "block"
-            "hint"
-          ]
-          ''
-            The behavior when `restricted_keys` trigger count mechanism.
-          '';
-
-      disabledKeys = helpers.defaultNullOpts.mkAttrsOf (with types; listOf str) {
-        "<Up>" = [
-          ""
-          "i"
-        ];
-        "<Down>" = [
-          ""
-          "i"
-        ];
-        "<Left>" = [
-          ""
-          "i"
-        ];
-        "<Right>" = [
-          ""
-          "i"
-        ];
-      } "Keys in what modes are disabled.";
-
-      disabledFiletypes = helpers.defaultNullOpts.mkListOf types.str [
-        "qf"
-        "netrw"
-        "NvimTree"
-        "lazy"
-        "mason"
-      ] "`hardtime.nvim` is disabled under these filetypes.";
-
-      hints =
-        helpers.mkNullOrOption
-          (
-            with types;
-            attrsOf (submodule {
-              options = {
-                message = lib.mkOption {
-                  description = "Hint message to be displayed.";
-                  type = helpers.nixvimTypes.rawLua;
-                };
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "hardtime";
+  originalName = "hardtime.nvim ";
+  package = "hardtime-nvim";
+
+  maintainers = [ lib.maintainers.what_name_should_go_here ];
+
```

Comment:
```suggestion
optionsRenamedToSettings = [
// TODO: add these 
]
```

Reply:
```nix
[
  "hint"
  "notification"
  "enabled"
  "hints"
  "maxTime"
  "maxCount"
  "disableMouse"
  "allowDifferentKey"
  "resettingKeys"
  "restrictedKeys"
  "restrictionMode"
  "disabledKeys"
  "disabledFiletypes"
]
```

Reply:
Although, passing through enable seems to be a mistake, or perhaps the plugin _does_ have an enable option, but we were previously using _our_ enable option (always `true`)?

Reply:
Looks like it does https://github.com/m4xshen/hardtime.nvim?tab=readme-ov-file#options

Reply:
Either way, we don't want to rename `plugins.hardtime.enable` -> `plugins.hardtime.settings.enable`, so we'll just have to make a breaking change here.

The plugin-default is `true` anyway, so in practice it shouldn't break anything.

Reply:
> Either way, we don't want to rename `plugins.hardtime.enable` -> `plugins.hardtime.settings.enable`.
> 
> The plugin-default is `true` anyway, so in practice it shouldn't break anything.

Sorry, I agree, was just surprised it had an enabled option haha